### DISCUSSION TITLE
m1 build in macports environment - issues with proper library path

### DISCUSSION THREAD

**Discussion by MStraeten** [GrandOrgue/grandorgue#1439] [Developers]
i'm trying to build grandorgue in my m1 macports environment and am failing to get the library path (-L /opt/local/lib) integrated in the generated link.txt files in `build/src/core/CMakeFiles`, `build/src/tools/CMakeFiles/GrandOrguePerfTest.dir` and `build/src/tools/CMakeFiles/GrandOrgueTool.dir`
it's fine in build/src/grandorgue/CMakeFiles/Grandorgue.dir

Any idea wich files needs to be edited to avoid adding it manually? 
```
cmake \
-DCMAKE_LIBRARY_PATH=/opt/local/lib \
```
doesn't help

**Comment by oleg68** [comment]
Why do you need /opt/local/lib?

The current cmake files use pkg-config for determining correct library directories.

**Reply by MStraeten** [reply]
it doesn't seem to properly take over the libwavpack library location (macports environment) 
in CMakecache it seems to be properly detected
[CMakeCache.txt](https://github.com/GrandOrgue/grandorgue/files/10951858/CMakeCache.txt)

[link.txt](https://github.com/GrandOrgue/grandorgue/files/10951863/link.txt)

**Comment by oleg68** [comment]
What does return

``pkg-config --libs wavpack``
?

**Comment by oleg68** [comment]
And the content of your `/usr/local/lib/pkgconfig/wavpack.pc`.

The mine is
```
prefix=/usr/local/Cellar/wavpack/5.6.0
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: wavpack
Description: wavpack library
Version: 5.6.0
Requires:
Conflicts:
Libs: -L${libdir} -lwavpack
Libs.private: 
Cflags: -I${includedir}
```

So the correct library path for wavpack should be in the `prefix` string.

**Reply by MStraeten** [reply]
```
pkg-config --libs wavpack
-L/opt/local/lib -lwavpack
```
```
more /opt/local/lib/pkgconfig/wavpack.pc
prefix=/opt/local
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: wavpack
Description: wavpack library
Version: 5.4.0
Requires:
Conflicts:
Libs: -L${libdir} -lwavpack
Libs.private:
Cflags: -I${includedir}
```

**Comment by MStraeten** [comment]
found a hacky way to force using the proper path:
```
cmake .. \
-DWAVPACK_LIBRARIES="-L/opt/local/lib -lwavpack" \
-DCMAKE_BUILD_TYPE=Release \
-G "Unix Makefiles"
```

**Comment by MStraeten** [comment]
quite interesting - never did that before, but GrandOrguePerfTest in official package contains a "hardcoded" reference to libwavpack:
```
Applications/GrandOrgue.app/Contents/MacOS/GrandOrguePerfTest
dyld[42317]: Library not loaded: /usr/local/opt/wavpack/lib/libwavpack.1.dylib
  Referenced from: <9F3B8891-CF7B-32AC-BBDF-59633F28706E> /Applications/GrandOrgue.app/Contents/MacOS/GrandOrguePerfTest
  Reason: tried: '/usr/local/opt/wavpack/lib/libwavpack.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/wavpack/lib/libwavpack.1.dylib' (no such file), '/usr/local/opt/wavpack/lib/libwavpack.1.dylib' (no such file), '/usr/local/lib/libwavpack.1.dylib' (no such file), '/usr/lib/libwavpack.1.dylib' (no such file, not in dyld cache)
zsh: abort      /Applications/GrandOrgue.app/Contents/MacOS/GrandOrguePerfTest
```
